index.js

```diff
@@ -51,9 +55,21 @@ function wrap(opts) {
 			return transform(input, metadata);
 		}
 
-		var hash = hashFn(input, metadata, salt);
+		var data = [ownHash || getOwnHash(), input];
+		if (salt) {
+			data.push(salt);
+		}
```

Comment:
Do you mean:

```js
		var data = [ownHash || getOwnHash(), input];

		if (salt) {
			data.push(salt);
		}
		if (hashData) {
			data = data.concat(hashData(input, metadata));
		}
```

Or:

```js
		var data = [ownHash || getOwnHash(), input];

		if (salt) {
			data.push(salt);
		}

		if (hashData) {
			data = data.concat(hashData(input, metadata));
		}
```

Cause there's more if-blocks to change in case of the latter. Which makes me thing this should wait until XO complains.